### PR TITLE
Add new CSS color syntax

### PR DESCRIPTION
**Pull request by MusiKid**
The commits add support for some CSS color syntax.

### CODE REVIEW

File: src/cli/parser.rs [hunk 1]
```diff
@@ -203,12 +283,22 @@ pub fn parse_color(color: &str) -> Option<Color> {
...
+        let ms = float_to_f64(caps.get(3).unwrap().as_str());
+        let ml = float_to_f64(caps.get(4).unwrap().as_str());
 
         match (mh, ms, ml) {
             (Some(h), Some(s), Some(l)) => {
-                let h = f64::from(h);
+                let h = match unit {
+                    "°" | "deg" => h,
+                    "grad" => h * (18.0 / 20.0),
+                    "rad" => h.to_degrees().round(), // Better to round it (because of PI being an irrational number)
```
Comment (sharkdp) [inline src/cli/parser.rs]: I don't think we should round here. That's just a loss of precision.

File: src/cli/parser.rs [hunk 2]
```diff
@@ -69,6 +69,46 @@ lazy_static! {
...
     .unwrap();
 
+    // rgb(255 0 119)
+    pub static ref RE_RGB_SPACE: Regex = Regex::new(
+        r"(?x)
+            ^
+            rgb\(
+                \s*
+                (\d{1,3})
+                \s*
```
Comment (sharkdp) [inline src/cli/parser.rs]: Shouldn't we have `\s+` as a separator between the numbers? Could you please add a test that `rgb(2550119)` is *not* parsed as a color?

File: src/cli/parser.rs [hunk 3]
```diff
@@ -287,11 +383,20 @@ fn parse_rgb() {
...
+    assert_eq!(Some(rgb(255, 0, 119)), parse_color("rgb(255 0 119)"));
+    assert_eq!(Some(rgb(255, 0, 119)), parse_color("rgb(    255      0      119)"));
 
     assert_eq!(None, parse_color("rgb(256,0,0)"));
     assert_eq!(None, parse_color("rgb(255,0)"));
     assert_eq!(None, parse_color("rgb(255,0,0"));
     assert_eq!(None, parse_color("rgb (256,0,0)"));
+    assert_eq!(None, parse_color("rgb (250%,240%,180%)"));
+    assert_eq!(None, parse_color("rgb (100%,0,0)"));
+    assert_eq!(None, parse_color("rgb (100%,555%,-45%)"));
```
Comment (sharkdp) [inline src/cli/parser.rs]: These three are not parsed as a number due to the space between `rgb` and `(`. However, it looks like you wanted to add tests for something else (too high or negative percentages?)

File: src/cli/parser.rs (hunk 1 above)
Comment (MusiKid) [reply src/cli/parser.rs]: Okay I'll keep it as it.

File: src/cli/parser.rs (hunk 2 above)
Comment (MusiKid) [reply src/cli/parser.rs]: Yes you're right!

File: src/cli/parser.rs (hunk 3 above)
Comment (MusiKid) [reply src/cli/parser.rs]: Yes I made some errors when I pasted :confused:.
